### PR TITLE
Fix regression WHERE (not) in (4.2.0)

### PR DESCRIPTION
**Pull request by paulrutter**
Fix for https://github.com/AlaSQL/alasql/issues/1829.

- Fix regression when using IN / NOT IN
- Using refs didn't work anymore
- Dropped set cache, as it's not really needed unless the same queries are performed often
- Added unit test to guard behavior

### CODE REVIEW

**Comment by paulrutter** [comment]
@mathiasrw Some regression introduced in 4.2.0, this PR fixes it.
I removed the Set cache as i couldn't get it to work with passing references. It would still be beneficial when running the same query with lots of IN values against different data though, but we might add that back in later. 

See the new unit test.

**Comment by mathiasrw** [comment]
Thank you so much for acting on this rapidly.

File: test/test1829.js
Comment (mathiasrw) [inline test/test1829.js:1]: Absolutely love it. Thank you!

**Comment by mathiasrw** [comment]
Released as part of 4.2.1

**Pull request merged**